src/nodes/rlp.rs

```diff
@@ -59,8 +65,11 @@ impl RlpNode {
 
     /// Creates a new RLP-encoded node from the given data.
     #[inline]
-    pub fn from_raw_rlp(data: &[u8]) -> alloy_rlp::Result<Self> {
-        Self::from_raw(data).ok_or(alloy_rlp::Error::Custom("RLP node too large"))
+    pub fn from_raw_rlp(data: &[u8]) -> Result<Self, TrieError> {
+        Self::from_raw(data).ok_or_else(|| TrieError::RlpNodeTooLarge {
+            size: data.len(),
+            max: MAX,
+        })
```

Comment:
makes more sense if this returns `alloy_rlp::Result<Self>` still and just do `TrieError::RlpNodeTooLarge.into()`